### ISSUE TITLE
Optional mapping error with ConfigMapping using Maps with enum keys, Nested Groups and Optional

### ISSUE THREAD

**Issue by rudiger3d** [quarkusio/quarkus#22334]
### Describe the bug

I'm seeing an error when mapping configuration to objects using a combination of nested groups (https://quarkus.io/guides/config-mappings#nested-groups), maps (https://quarkus.io/guides/config-mappings#maps) with Enums as key and conditional mapping (https://smallrye.io/docs/smallrye-config/mapping/mapping.html#_optional_config_mapping) that I think is an (obscure) bug.

The problem I'm seeing appears to be somewhat related to https://github.com/quarkusio/quarkus/issues/21407, because the same "key" used in different part levels of the config hierarchy somehow wreaks havoc. This time the example also involves conditional mapping and using Enums as keys maps.

First I'm giving a short example with some code that is commented out. The test in this example works. 

When re-inserting the commented out code, the test in the example fails. (Note that I'm only experiencing the problem when using enums as keys in the map - not with Strings)

Having

```java
@ConfigMapping(prefix = "clients", namingStrategy = ConfigMapping.NamingStrategy.VERBATIM)
public interface ClientsConfiguration {
    enum ClientId { SOS_DAH, NAF }

    @WithParentName
    Map<ClientId, ClientConfiguration> clients();

    interface ClientConfiguration {
        Optional<CreatedByProperties> app();
        CreatedByProperties web();
//        MediumProperties medium();
    }

    interface CreatedByProperties {
        String createdByApplication();
    }

//    interface MediumProperties {
//        boolean web();
//        @WithDefault("false")
//        boolean app();
//    }
}
```

and application.properties:

```
clients.SOS_DAH.web.createdByApplication=RoadrunnerWeb
clients.SOS_DAH.app.createdByApplication=Roadrunner
#clients.SOS_DAH.medium.web=true
#clients.SOS_DAH.medium.app=true

clients.NAF.web.createdByApplication=RoadrunnerWebNAF
#clients.NAF.medium.web=true
```

A resource:
```java
@Path("/greeting")
public class GreetingResource {
    @Inject
    ClientsConfiguration clientsConfiguration;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return clientsConfiguration.clients().get(ClientId.SOS_DAH).app().get().createdByApplication();
    }
}
```

and a test:

```java
@QuarkusTest
public class GreetingResourceTest {
    @Inject
    ClientsConfiguration clientsConfiguration;

    @Test
    public void testHelloEndpoint() {
        given()
                .when().get("/greeting")
                .then()
                .statusCode(200)
                .body(is("Roadrunner"));
    }
}
```

The test runs **successfully**.

When re-inserting the commented out code, i.e. removing the // in the ClientsConfiguration interface and the # in the application.properties, the test now fails with:

```
2021-12-17 21:58:14,428 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-0) HTTP Request to /greeting failed, error id: dc908705-d552-4525-ab95-f15938f3d64c-1: org.jboss.resteasy.spi.UnhandledException: java.util.NoSuchElementException: No value present
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:105)
<OMITTED>
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:143)
	at org.acme.config.GreetingResource.hello(GreetingResource.java:31)
<OMITTED>
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	... 15 more
```

### Expected behavior

The same key used in different part levels of the config hierarchy, should not affect conditional mapping (when using enums).

### Actual behavior

The same key used in different part levels of the config hierarchy affects conditional mapping (when using enums).

### How to Reproduce?

1. git clone https://github.com/quarkusio/quarkus-quickstarts/tree/main/config-quickstart
2. cd config-quickstart
3. Add ClientsConfiguration.java and the properties from the "Describe the error" section to application.properties
4. Modify GreetingResource.java and GreetingResourceTest.java as per the description above.
5. Run the GreetingResourceTest.java

### Output of `uname -a` or `ver`

Darwin everythingsruined.local 20.6.0 Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk 17 2021-09-14 OpenJDK Runtime Environment Temurin-17+35 (build 17+35) OpenJDK 64-Bit Server VM Temurin-17+35 (build 17+35, mixed mode, sharing)

### Additional information

Tagging @radcortez as he fixed the bugs that I previously submitted :)

**Comment by radcortez** [comment]
I'll have a look

**Comment by radcortez** [comment]
This will be fixed by https://github.com/smallrye/smallrye-config/pull/685

**Comment by rudiger3d** [comment]
Thanks @radcortez

**Comment by radcortez** [comment]
I've released SR Config 2.8.0. You should be able to update it manually to get the fix (or wait for the next Quarkus release).
